SVA/lib/handlers.S

```diff
@@ -507,24 +507,58 @@ SVAsyscall:
    * if this is the fork(), vfork(), rfork(), or pdfork() system calls.  The
    * system call number will be in %rax and can have the following values
    * (you can find these in syscalls.master) within the FreeBSD kernel source
-   * code):
+   * code).
+   * According to the type of fork and the flags, we make a decision about
+   * whether to COW on ghost memory or not, and save the decision in the flags
+   * filed of the current SVAThread. (1 is COW, and 0 is not COW.)
+   * In particular, fork() and pdfork() always incur COW, vfork() does not incur COW,
+   * and for rfork() it depends on its arguments.
    *
    *   2 - fork()
    *  66 - vfork()
    * 251 - rfork()
    * 518 - pdfork()
    */
+
+  /* Find the flags field of the current SVAThread */
+  movq %gs:0x260, %r14          
+  movq CPU_THREAD(%r14), %r14    
+  movq $0,  0x7d80(%r14)  
```

Comment:
You should create a macro for the 0x7d80 constant.  I assume it's some offset into a C structure.